lib/platform/bitbucket/index.js

```diff
@@ -0,0 +1,86 @@
+const bbGot = require('./bb-got-wrapper');
+
+module.exports = {
+  // Initialization
+  getRepos,
+  initRepo,
+
+  // Search
+  getFileList,
+
+  // More will follow
+
+  // misc
+  getPrList,
+};
+
+// Get all repositories that the user has access to
+async function getRepos() {
+  /**
+   * `role`
+   * values:
+   * - admin—explicit administrator access
+   * - contributor—explicit write access
+   * - member—explicit read access
+   * - owner—owned by the current user
+   * @see https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D
+   */
+  /**
+   * `pagelen`
+   * Current number of objects on the existing page.
+   * The default value is 10 with 100 being the maximum allowed value.
+   * @see https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories
+   */
+  const { body } = await bbGot('/repositories/?role=contributor&pagelen=100');
+
+  return body.values.map(x => x.full_name);
+}
+
+// Initialize GitHub by getting base branch and SHA
+async function initRepo(repoFullName) {
+  const { body } = await bbGot(`/repositories/${repoFullName}`);
+  const repo = {};
+  repo.repoFullName = repoFullName;
+  repo.mainbranch = body.mainbranch.name;
+  repo.private = body.is_private;
+  repo.fork = !!body.parent;
+  /**
+   * mergeMethod is only merge for BB cloud
+   * @see https://bitbucket.org/site/master/issues/8995/provide-the-option-to-use-git-merge-squash
+   */
+  repo.mergeMethod = 'merge';
+  const prList = await getPrList(repoFullName);
+  return { repo, prList };
+}
+
+// Search
+
+// Get full file list
+async function getFileList(repoFullName, branchName) {
+  // bitbucket doesnt support recursion for this endpoint
+  // https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/src/%7Bnode%7D/%7Bpath%7D
+  const res = await bbGot(`/repositories/${repoFullName}/src`);
+  // const values = res.body.values;
+
+  // TODO: add support for recursive listing
```

Comment:
That's a real pain that BitBucket doesn't support any recursive file listing. It's pretty essential that Renovate can get a full list of:
- files
- Pull Requests
- Branches

Reply:
@iamstarkov check out https://bitbucket.org/!api/1.0/repositories/iamstarkov/phenomic/directory/master :) 

Found it mentioned here: https://bitbucket.org/site/master/issues/8316/rest-api-list-all-files-in-repo-like-git

You'll need to adjust your bb wrapper somehow to allow this URL to be queried but I think it's worth it.

Reply:
isnt it the 1st version of the api? i though it will be deprecated

Reply:
@rarkins  do we need full file system tree of the repo for the monorepos?

Reply:
Full file system would be necessary for monorepos only. So in theory you could skip it in initial release if you list monorepos as unsupported. However I'd prefer to try/catch the API v1 and fall back to API v2 with a warning, as that way you gain something with essentially zero risk of the API gets revoked before they add recursive support to latest API

Reply:
i think we can reduce amount of api calls, by detecting monorepos and executing recursive scheme only for those repos

Reply:
and i think its applicable for other platforms as well. maybe not in terms of API hits, but in terms of network bandwidth and latency/speed